### ISSUE TITLE
[java] UnusedPrivateMethod false positive when outer class calls private static method on inner class

### ISSUE THREAD

**Issue by sratz** [pmd/pmd#3468]
**Affects PMD Version:** >=6.33.0

**Rule:** [UnusedPrivateMethod](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:** Outer class calling a private method of a static inner class. -> False positive ('unused private method') on the inner class method.

Looks like this was introduced by #3157.

**Code Sample demonstrating the issue:**

```java
public class OuterClass {

	public void foo() {
		InnerClass.doSomething();
	}

	static class InnerClass {
		private static void doSomething() {
		}
	}
}
```

**Expected outcome:** No PMD messages

PMD reports a violation at line 8, but that's wrong. That's a false positive.

**Running PMD through:** *Maven | [eclipse-pmd](https://github.com/eclipse-pmd/eclipse-pmd/)*
